nomad/state/schema.go

```diff
@@ -91,6 +91,15 @@ func jobTableSchema() *memdb.TableSchema {
 					Lowercase: true,
 				},
 			},
+			"type": &memdb.IndexSchema{
+				Name:         "type",
+				AllowMissing: false,
+				Unique:       false,
+				Indexer: &memdb.StringFieldIndex{
+					Field:     "Type",
+					Lowercase: false,
```

Comment:
Scheduler types probably should be case-insensitive, even though its mostly hard-coded everywhere we probably shouldn't open this up. Is there a reason we would want this to be case-sensitive?